ecommerce/serializers.py

```diff
@@ -519,11 +535,20 @@ def validate_items(self, items):
 
             if product_id is None:
                 raise ValidationError("Invalid request")
-            if not models.ProductVersion.objects.filter(
-                product__id=product_id
-            ).exists():
+
+            try:
+                product = get_product_from_querystring_id(
+                    models.Product.all_objects, product_id
+                )
+            except (ObjectDoesNotExist, MultipleObjectsReturned) as exc:
+                if isinstance(exc, MultipleObjectsReturned):
+                    log.error(
+                        "Multiple Products found with identical ids: %s", product_id
+                    )
+                raise ValidationError(f"Invalid product id {product_id}") from exc
+
+            if not product:
```

Comment:
This validation (`if not product`) is no longer necessary due to the `get_product_from_querystring_id` exception handling, and the one below it (`if not product.is_active`) was never really necessary. We certainly don't need to be telling end users that a product is inactive instead of just telling them that the product id is invalid. Given that, I think we should only pass `product_id` into `get_product_from_querystring_id`, and just use `Product.objects` inside that function

Reply:
@gsidebo We are already showing the inactive validation error. @asadiqbal08 can you confirm do we really need to change that?

Reply:
Yeah, that validation was already in the code base before but not sure why. 
I agree we should not tell the end users about the product active/inactive state.

Reply:
Yeah I know the inactive product validation was existing behavior, but I'm certain that we don't need it. The concept of active/inactive products is exclusively an engineering concern. If we see a log message or get a support ticket about an error message that says "Inactive product id 5" vs. "Invalid product id 5", the "Inactive" message is not any more helpful to us, and it certainly isn't more helpful to the user. We should change this and make the behavior consistent, which is to say we should just use `Product.objects` in that helper function

Reply:
@gsidebo what your thoughts about it?